src/org/zaproxy/zap/extension/exportReport/Export/ExportReport.java

```diff
@@ -0,0 +1,255 @@
+ * CREATED ON	:	MARCH 10TH, 2016
+ * CURRENT VER	:	V1.0
+ * SOURCE		:	https://github.com/JordanGS/workspace/tree/master/zap-extensions/src/org/zaproxy/zap/extension/exportReport
+ */
+
+/* 
+ * MODIFED BY	:	<NAME> - <GIT USER>
+ * MOD DATE		:	
+ * MOD VERSION	:	<VERSION OF PLUGIN>
+ * MOD DESC		:	
+ */
+
+public class ExportReport {
+	private Logger logger = Logger.getLogger(ExportReport.class);
+
+	private Task task; // Create a task and dump the export into the background,
+						// essential for large exports.
+	private WritableFileChooser fc; // Global because the property change
+									// listener won't be able to access if it's
+									// local scope.
+
+	class Task extends SwingWorker<Void, Void> {
+		private ViewDelegate view;
+		private ExtensionExportReport extension;
+		private String fileExtension;
+
+		public Task(ViewDelegate view, ExtensionExportReport extension, String fileExtension) {
+			this.view = view;
+			this.extension = extension;
+			this.fileExtension = fileExtension;
+		}
+
+		/**
+		 * Main task. Executed in background thread.
+		 */
+		@Override
+		public Void doInBackground() {
+			File f = fc.getSelectedFile();
+
+			String file = f.getName();
+			String absolutePath = f.getAbsolutePath();
+			String fileName = file.substring(0, file.lastIndexOf(fileExtension) - 1);
+			String path = absolutePath.substring(0, absolutePath.lastIndexOf(file));
+
+			String XML_PATH = "";
+			try
+			{
+				XML_PATH = ReportExport.generateDUMP(path, fileName, extension.extensionGetTitle(),
+						extension.extensionGetBy(), extension.extensionGetFor(), extension.extensionGetScanDate(),
+						extension.extensionGetScanVer(), extension.extensionGetReportDate(),
+						extension.extensionGetReportVer(), extension.getDescription(),
+						extension.getIncludedAlertSeverity(), extension.getIncludedAlertDetails());
+			}
+			catch (UnsupportedEncodingException e)
+			{
+				logger.error(e.getMessage(), e);
+				view.showMessageDialog(MessageFormat.format("Error: Problem encoding the DUMP File content, please see log for more details.", new Object[] { absolutePath }));
+			}
+
+			String extensionPath = Constant.getZapHome() + "xml" + File.separator;
+			String mergeXSL = (extensionPath + "merge.xml.xsl");
+			String reportXSL = (extensionPath + "report.html.xsl");
+
+			File f_view = null;
+			boolean show = false;
+			switch (fileExtension.toLowerCase(Locale.ROOT)) {
+			case Utils.html:
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
+				f_view = null;
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, absolutePath,(path + fileName + ".xml"), reportXSL);
+				deleteFile(path + fileName + ".xml", "The merged XML file: ");
+				show = true;
+				break;
+			case Utils.bootstrap:
+				view.showMessageDialog(MessageFormat.format("Bootstrap: Currently unavilable, expected release is v2.0.", new Object[] { absolutePath }));
```

Comment:
Oversight on my part, i didn't want to remove the option since i plan on releasing a bootstrap export option so i wanted to show a dialog message.